src/Components/Quiz.tsx

```diff
@@ -1,86 +1,158 @@
 import http from "../Helpers/http";
 import Loader from "./Loader";
 import "./Style.css";
 import Countdown from "react-countdown";
+import dotenv from "dotenv"
+
+const CryptoJS = require("crypto-js");
 
 const Quiz = () => {
+
+  const [text, setText] = useState("");
+  const [screen, setScreen] = useState("encrypt");
+
+  const [encrptedData, setEncrptedData] = useState("");
+  const [decrptedData, setDecrptedData] = useState("");
+
   const [success, setSuccess] = useState(0);
   const [fail, setFail] = useState(0);
   const [questionIndex, setQuestionIndex] = useState(0);
-  const [optionButton, setOptionButton] = useState(false);
   const [nextButton, setNextButton] = useState(true);
   const [showResults, setShowResults] = useState(false);
+  const [selectedOption, setSelectedOption] = useState("")
   const [score, setScore] = useState(0);
-  const [countdown, setCountdown] = useState(30);
   const [progressBar, setProgressBar] = useState(10);
   const [checkedBox, setCheckedBox] = useState<any>(false);
   const [data, setData] = useState<any>([]);
+  const [progressBarIncrement,setProgressBarIncrement] = useState<any>()
   const navigate = useNavigate();
   const [time, setTime] = useState(30);
+  const [checkedState,setCheckedState] = useState(false)
+  // const secretPass = "XkhZG4fW2t2W";
+  const secretPass = process.env.REACT_APP_SECRETPASS
+  
 
   useEffect(() => {
     const intervalId = setInterval(() => {
+      // console.log("inside interval fus");
+      
       if (time == 0) {
         clearInterval(intervalId);
+        // console.log("interval stopped");
+        
       } else {
+        // console.log(setTime);
+        // console.log("interval running");
+        
         setTime((prevTime) => prevTime - 1);
-  
       }
     }, 1000);
 
     return () => clearInterval(intervalId);
-  }, []);
+  },)
+
+ 
 
+
+  const nextFunction =()=>{
+    calculateScore()
+    if (questionIndex + 1 == data.length) {
+      console.log(questionIndex, "Index");
+      console.log("end of questions");
+      setShowResults(true);
+      setTime(0);
+      setSelectedOption('')
+      let res = ((success / data.length) * 100).toString().slice(0,4)
+      setScore(Number(res));
+
+    } else {
+      setQuestionIndex(questionIndex + 1);
+      setNextButton(true);
+      setTime(30);
+      setProgressBar(progressBar + progressBarIncrement);
+      setSelectedOption('')
+    }
+    resetOption();
+  
+  }
   useEffect(() => {
     if (time === 0) {
       // Handle timer completion
-      if (questionIndex + 1 == data.length) {
-        console.log(questionIndex, "Index");
-        console.log("end of questions");
+      if (questionIndex + 1 == data.length && selectedOption=='') {
+        console.log(questionIndex, "Index when over");
+        setFail(fail + 1);
         setShowResults(true);
-        setScore((success / (success + fail)) * 100);
+        let res = ((success / data.length) * 100).toString().slice(0,4)
```

Comment:
You wanted `xx.xx` as result? `
```suggestion
        const score = ((success / data.length) * 100).toFixed(2)
```